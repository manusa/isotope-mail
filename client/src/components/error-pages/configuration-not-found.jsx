import React from 'react';
import {translate} from 'react-i18next';
import mainCss from '../../styles/main.scss';

export const ConfigurationNotFound = ({t}) => (
  <div className={mainCss['isotope-error-page']}>
    <div className={mainCss['isotope-error-page__overlay']}>
      <div className={mainCss['isotope-error-page__container']}>
        <h1>{t('configurationNotFound.Sorry')}</h1>
        <p>{t('configurationNotFound.message')}</p>
        <a
          className={
            `${mainCss['mdc-button']} ${mainCss['mdc-button--outlined']} ${mainCss['isotope-error-page__button']}`}
          href={'/'}>{t('configurationNotFound.Retry')}</a>
      </div>
    </div>
  </div>
);

export default translate()(ConfigurationNotFound);
