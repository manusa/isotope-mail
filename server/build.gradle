buildscript {
	ext {
		springBootVersion = '2.0.4.RELEASE'
	}
	repositories {
		mavenCentral()
	}
	dependencies {
		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
	}
}

apply plugin: 'java'
apply plugin: 'eclipse'
apply plugin: 'org.springframework.boot'
apply plugin: 'io.spring.dependency-management'

group = 'com.marcnuri.isotope'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = 1.8

repositories {
	mavenCentral()
}


dependencies {
	compile('org.springframework.boot:spring-boot-starter')
	compile('org.springframework.boot:spring-boot-starter-web')
	compile('org.springframework.boot:spring-boot-starter-hateoas')
	compile('com.sun.mail:javax.mail:1.6.1')
	testCompile('org.springframework.boot:spring-boot-starter-test')
	testCompile('com.marcnuri:spring-common-test:0.0.2')
}
