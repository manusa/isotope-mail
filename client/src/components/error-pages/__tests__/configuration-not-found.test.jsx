import React from 'react';
import {shallow} from 'enzyme/build/index';
import {ConfigurationNotFound} from '../configuration-not-found';


describe('ConfigurationNotFound component test suite', () => {
  test('Snapshot render, should render ConfigurationNotFound', () => {
    // Given
    const props = {t: key => key};
    // When
    const configurationNotFound = shallow(<ConfigurationNotFound {...props}/>);
    // Then
    expect(configurationNotFound).toMatchSnapshot();
  });
});
