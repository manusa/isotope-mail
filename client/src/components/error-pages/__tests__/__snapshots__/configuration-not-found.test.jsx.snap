// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationNotFound component test suite Snapshot render, should render ConfigurationNotFound 1`] = `
<div
  className="isotope-error-page"
>
  <div
    className="isotope-error-page__overlay"
  >
    <div
      className="isotope-error-page__container"
    >
      <h1>
        configurationNotFound.Sorry
      </h1>
      <p>
        configurationNotFound.message
      </p>
      <a
        className="mdc-button mdc-button--outlined isotope-error-page__button"
        href="/"
      >
        configurationNotFound.Retry
      </a>
    </div>
  </div>
</div>
`;
